modules/apps/object/object-test/src/testFunctional/tests/ObjectRelationships.testcase

```diff
@@ -266,6 +266,147 @@ definition {
+
+			ObjectAdmin.addObjectFieldViaAPI(
+				fieldBusinessType = "Text",
+				fieldLabelName = "Custom Field",
+				fieldName = "customObjectField",
+				fieldType = "String",
+				isRequired = "false",
+				objectName = "CustomObject");
+
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObject",
+				objectName_2 = "User",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "manyToMany");
+
+			ObjectAdmin.publishObjectViaAPI(objectName = "CustomObject");
+		}
+
+		task ("Add layout with relationship tab") {
+			ObjectAdmin.addObjectLayoutViaAPI(
+				layoutName = "Custom Layout",
+				objectName = "CustomObject");
+
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object");
+
+			ObjectAdmin.goToLayoutsTab();
+
+			ObjectAdmin.selectSpecificKebabMenuOption(
+				kebabOption = "View",
+				objectLabel = "Custom Layout");
+
+			ObjectAdmin.markLayoutAsDefault();
+
+			ObjectAdmin.goToLayoutTabOnLayouts();
+
+			ObjectAdmin.addTabFieldsOnLayout(
+				blockName = "Block 1",
+				columnsNumber = "1",
+				fieldLabelName = "Custom Field",
+				tabName = "Field Tab");
+
+			ObjectAdmin.addTabRelationshipsOnLayout(
+				relationshipsLabelName = "Relationship",
+				tabName = "Relationship Tab");
+
+			Button.clickSave();
+		}
+
+		task ("Define First Name as a title field on the native object") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectCustomObject(label = "User");
+
+			CreateObject.selectTitleField(fieldLabel = "First Name");
+
+			CreateObject.saveObject();
+		}
+
+		task ("Add 2 entries by API to relate to user entries") {
+			for (var entryName : list "A,B") {
+				ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+					fieldName = "customObjectField",
+					objectName = "CustomObject",
+					value = "Entry ${entryName}");
+			}
+		}
+
+		task ("Relates custom object entries to user object entries") {
+			Navigator.openURL();
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject");
+
+			for (var entryName : list "A,B") {
+				ObjectPortlet.viewEntryDetails(entry = "Entry ${entryName}");
+
+				ObjectAdmin.gotoRelationshipsTab();
+
+				CreateObject.addRelationshipAndSelectEntry(titleFieldName = "userfn_1");
+
+				CreateObject.addRelationshipAndSelectEntry(titleFieldName = "userfn_2");
+
+				ObjectPortlet.viewEntry(entry = "userfn_1");
+
+				ObjectPortlet.viewEntry(entry = "userfn_2");
+
+				Navigator.gotoBack();
+			}
+		}
+
+		task ("Delete relationship between a custom object entry to native object entry") {
+			for (var entryName : list "A,B") {
+				ObjectPortlet.viewEntryDetails(entry = "Entry ${entryName}");
+
+				CreateObject.deleteEntryInRelationshipTab(entryName = "userfn_1");
+
+				CreateObject.deleteEntryInRelationshipTab(entryName = "userfn_2");
+
+				ObjectPortlet.assertEntryNotPresent(entry = "userfn_1");
+
+				ObjectPortlet.assertEntryNotPresent(entry = "userfn_2");
```

Comment:
Ideally, the flow here should be:
1. Delete Relationship A
2. Assert Relationship A is deleted
3. Delete Relationship B
4. Assert Relationship B is deleted

If we delete both relationships then assert they're deleted, we're not actually checking if the act of deleting Relationship A did actually delete it. It could be that Relationship A wasn't deleted and deleting Relationship B actually deleted both Relationships.